PhysicsTools/PatAlgos/python/slimming/enableDRN.py

```diff
@@ -0,0 +1,17 @@
+import FWCore.ParameterSet.Config as cms
+from RecoEgamma.EgammaTools.egammaObjectModificationsInMiniAOD_cff import egamma_modifications
+
+def customise(process):
+  process.load("PhysicsTools.PatAlgos.slimming.patPhotonDRNCorrector_cfi")
+  process.DRNTask = cms.Task(process.patPhotonsDRN)
+  process.schedule.associate(process.DRNTask)
+  egamma_modifications.append(
```

Comment:
Modifying python objects defined in other files is fragile (and generally forbidden by policy). For example, assuming this is the main user of `RecoEgamma.EgammaTools.egammaObjectModificationsInMiniAOD_cff.egamma_modifications`
https://github.com/cms-sw/cmssw/blob/ce4f5b212642a02e5c7ccff8403fdbf6f0306865/PhysicsTools/PatAlgos/python/slimming/miniAOD_tools.py#L287-L289
if it would be used here as `egamma_modifications.clone()`, which is the recommended pattern, the modification above would not have the desired effect (*). It would be clearer to customize the intended modules explicitly.

(*) on the other hand, using `egamma_modifications.clone()` would break also
https://github.com/cms-sw/cmssw/blob/ce4f5b212642a02e5c7ccff8403fdbf6f0306865/PhysicsTools/PatAlgos/python/slimming/miniAOD_tools.py#L345-L347

Reply:
Would the recommended solution be to instead move everything into an modifier rather than the customization function? The reason I'm currently using a customization function is that I couldn't figure out how to have a modifier add a task to the process in addition to modifying the egamma_modifications PSet. Is there a way to do this?

Reply:
A modifier can be used to add a module or a Task into a Task or Sequence (that that end up included into a Path, that is included in the Schedule if the Process has one), see https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideCmsDriverEras#Replacing_whole_objects.

In general for use in production a modifier is preferred over customize functions, whereas customize functions are more suitable for test setups (or some other specific cases).

Reply:
Ah okay, thank you for the help. I will update my code to use a modifier rather than the customization function in the way you suggest.